File: frontend/src/types/photo.ts
/**
 * 照片相关类型定义
 */

// AI 标签
export interface AITags {
  subject_emotion?: string
  pose?: string
  clothing_style?: string
  lighting?: string
  environment?: string
  overall_description?: string
  aesthetic_score?: number
  deep_analysis?: string
  deep_analysis_time?: string
}

// EXIF 信息
export interface ExifInfo {
  shot_time?: string
  camera_model?: string
  lens_model?: string
  focal_length?: number
  aperture?: number
  shutter_speed?: string
  iso?: number
  width?: number
  height?: number
  orientation?: number
}

// 照片缩略图 URL
export interface PhotoUrls {
  original_url?: string
  thumbnail_small_url?: string
  thumbnail_medium_url?: string
  thumbnail_large_url?: string
}

// 照片基础信息
export interface Photo {
  id: string
  filename: string
  original_path: string
  file_size: number
  file_hash?: string
  
  // EXIF信息
  shot_time?: string
  camera_model?: string
  lens_model?: string
  focal_length?: number
  aperture?: number
  shutter_speed?: string
  iso?: number
  
  // 缩略图路径
  thumbnail_small?: string
  thumbnail_medium?: string
  thumbnail_large?: string
  
  // 完整URL（COS或本地）
  original_url?: string
  thumbnail_small_url?: string
  thumbnail_medium_url?: string
  thumbnail_large_url?: string
  
  // AI标签
  ai_tags?: AITags
  
  // 用户数据
  rating: number
  is_favorite: boolean
  face_cluster_id?: string
  
  created_at: string
  updated_at: string
}

// 照片列表响应
export interface PhotoListResponse {
  photos: Photo[]
  total: number
  page: number
  page_size: number
}

// 照片筛选选项
export interface PhotoFilterOptions {
  cameras: string[]
  years: number[]
  min_rating: number
  max_rating: number
}

// 更新照片请求
export interface UpdatePhotoRequest {
  rating?: number
  is_favorite?: boolean
  ai_tags?: Partial<AITags>
}

// 批量操作请求
export interface BatchDeleteRequest {
  photo_ids: string[]
}

// 批量操作响应
export interface BatchOperationResponse {
  success: string[]
  failed: Array<{ id: string; reason: string }>
}

// 上传结果
export interface UploadResult {
  filename: string
  photo_id?: string
  status: 'success' | 'skipped' | 'error'
  reason?: string
}

// 上传响应
export interface UploadResponse {
  uploaded: number
  skipped: number
  files: UploadResult[]
}

File: backend/app/schemas/photo.py
from pydantic import BaseModel
from typing import Optional, Dict
from datetime import datetime


class AITags(BaseModel):
    """AI标签模型"""
    subject_emotion: Optional[str] = None
    pose: Optional[str] = None
    clothing_style: Optional[str] = None
    lighting: Optional[str] = None
    environment: Optional[str] = None
    overall_description: Optional[str] = None
    aesthetic_score: Optional[float] = None
    deep_analysis: Optional[str] = None
    deep_analysis_time: Optional[str] = None


class PhotoBase(BaseModel):
    """照片基础模型"""
    filename: str
    file_size: int
    shot_time: Optional[datetime] = None
    camera_model: Optional[str] = None
    lens_model: Optional[str] = None
    focal_length: Optional[float] = None
    aperture: Optional[float] = None
    shutter_speed: Optional[str] = None
    iso: Optional[int] = None


class PhotoCreate(PhotoBase):
    """照片创建模型"""
    original_path: str
    file_hash: str


class PhotoResponse(PhotoBase):
    """照片响应模型"""
    id: str
    original_path: Optional[str] = None
    thumbnail_small: Optional[str] = None
    thumbnail_medium: Optional[str] = None
    thumbnail_large: Optional[str] = None
    ai_tags: Optional[AITags] = None
    rating: int = 0
    is_favorite: bool = False
    face_cluster_id: Optional[str] = None
    created_at: datetime
    updated_at: datetime
    
    class Config:
        from_attributes = True


class PhotoListResponse(BaseModel):
    """照片列表响应模型"""
    photos: list[PhotoResponse]
    total: int
    page: int
    page_size: int


class PhotoUpdate(BaseModel):
    """照片更新模型"""
    rating: Optional[int] = None
    is_favorite: Optional[bool] = None
